### PR TITLE
fix: controlled text input

### PR DESCRIPTION
**Pull request by seahorsepip**
Use value from props instead of local state for controlled `TextInput`.

### Summary

See #3144

### Test plan

Test both a controlled and non controlled input, check if value shown in input is correct.

### CODE REVIEW

File: src/components/TextInput/TextInput.tsx [hunk 1]
```diff
@@ -220,9 +220,12 @@ const TextInput = React.forwardRef<TextInputHandles, TextInputProps>(
     const [placeholder, setPlaceholder] = React.useState<string | undefined>(
       ''
     );
-    const [value, setValue] = React.useState<string | undefined>(
-      validInputValue
-    );
+    let [value, setValue] = React.useState<string | undefined>(validInputValue);
+    if (isControlled) {
+      // Use value from props instead of local state when input is controlled
+      value = rest.value;
```
Comment (drplauska) [inline src/components/TextInput/TextInput.tsx:226]: useState value shouldn't be reassigned like this.

Comment (drplauska) [inline src/components/TextInput/TextInput.tsx:226]: useState value shouldn't be reassigned like this.

Comment (seahorsepip) [reply src/components/TextInput/TextInput.tsx]: I don't think it makes much of a difference but I've changed the code t use a separate variable `value`, the state `value` is renamed to `uncontrolledValue`.

File: src/components/TextInput/TextInput.tsx [hunk 2]
```diff
@@ -220,9 +220,10 @@ const TextInput = React.forwardRef<TextInputHandles, TextInputProps>(
     const [placeholder, setPlaceholder] = React.useState<string | undefined>(
       ''
     );
-    const [value, setValue] = React.useState<string | undefined>(
-      validInputValue
-    );
+    let [uncontrolledValue, setUncontrolledValue] = React.useState<string | undefined>(validInputValue);
```
Comment (lukewalczak) [inline src/components/TextInput/TextInput.tsx:223]:
```suggestion
    const [uncontrolledValue, setUncontrolledValue] = React.useState<
      string | undefined
    >(validInputValue);
```

**Review by lukewalczak — APPROVED** [review]

**Pull request merged**